code/datums/components/spill.dm

```diff
@@ -0,0 +1,56 @@
+// This component is for forcing strange things into your pocket that fall out if you fall down
+// Yes this exists purely for the spaghetti meme
+
+/datum/component/spill
+	var/preexisting_item_flags
+
+	var/list/droptext
+	var/list/dropsound
+
+// droptext is an arglist for visible_message
+// dropsound is a list of potential sounds that gets picked from
+/datum/component/spill/Initialize(list/_droptext, list/_dropsound)

```

Comment:
You should always use underline after the parameter name to avoid future name clashes (hidden globals or platform specific things will have a underline before the name)

Reply:
Byond doesn't name things like that and neither do we for globals/internals